components/region_cache_memory_engine/src/engine.rs

```diff
@@ -222,11 +224,55 @@ impl RangeCacheMemoryEngine {
     pub fn evict_range(&mut self, range: &CacheRange) {
         let mut core = self.core.write().unwrap();
         if core.range_manager.evict_range(range) {
+            // todo: schedule it to a separate thread
             core.engine.delete_range(range);
+            core.mut_range_manager().on_delete_range(range);
+        }
+    }
+
+    pub(crate) fn handle_pending_load(&self) {
```

Comment:
It's not efficient to blindly call write lock regardless if there's work to do or not. 
Instead, we can quickly check  with read lock (ideally not lock at all but rust may not allow that) if there's any work to do. If there's work to do, acquire the write lock.